src/controller/GameController.ts

```diff
@@ -70,37 +71,52 @@ export class GameController {
         return this.gameState;
     }
 
-    public respondToEvent(responseId: String): CompletedEvent {
+    public respondToEvent(responseId: String): EndOfTurnSummary {
         const response = this.storyEvents.flatMap((it) => it.responses).find((it) => it.id === responseId);
         if (!response) {
             throw new Error(`Cannot find response with id: ${responseId}`);
         }
-
         if (!response.isApplicable(this.gameState)) {
             throw new Error('Response is not applicable');
         }
         const result = response.onSelect(this.gameState);
-        this.playerActionsForTurn.capabilityImprovements = this.playerActionsForTurn.capabilityImprovements.concat(
-            result.playerActions.capabilityImprovements
-        );
-        this.playerActionsForTurn.containmentPolicies = this.playerActionsForTurn.containmentPolicies.concat(
-            result.playerActions.containmentPolicies
-        );
 
+        // Get this current event being responded to
+        const thisEvent = this.storyEvents.find((it) => it.id === response.eventId);
+        
         // retain all events that are not the one that the current response pertains to
         this.eventsToRespond = this.eventsToRespond.filter((evt) => evt.id !== response.eventId);
-
+        
         // Update the game state
         this.gameState.indicators = result.updatedIndicators;
+        this.gameState.indicators.containmentPolicies = result.playerActions.containmentPolicies;
+        this.gameState.indicators.support = Math.min(100, Math.max(0, 50 - result.updatedIndicators.poll.support)); // Calculate support
+        this.playerActionsForTurn.containmentPolicies = result.playerActions.containmentPolicies;
```

Comment:
Indicators and containment policies set by the player's response to the event. Then fed through to the simulator, this is clunky- simulator should be integrated into the controller so it uses the controller's properties instead of having and settings its own set of different properties